### PR TITLE
Add state parameter to sign_in_with_apple request

### PR DESCRIPTION
**Pull request by HenriBeck**
#77

### CODE REVIEW

**Comment by HenriBeck** [comment]
@tp feel free to have a look here. I didn't have time to test this yet, I hope I can find some time later this week to do this.

File: packages/sign_in_with_apple/lib/src/sign_in_with_apple.dart [hunk 1]
```diff
@@ -214,6 +222,9 @@ class SignInWithApple {
 
         if (nonce != null)
           'nonce': nonce,
+
+        if (state != null)
+          'state': 'state',
```
Comment (tp) [inline packages/sign_in_with_apple/lib/src/sign_in_with_apple.dart:227]:
```suggestion
          'state': state,
```

**Comment by tp** [comment]
Tested this successfully on iOS and Android:

## iOS

client:
image

server:
image



## Android

with `state` and JWT

image


@HenriBeck Please have a look at the changes, and publish & tag if fine with you 🙏

**Comment by HenriBeck** [comment]
@tp looks good. Thanks for the clean up here

**Pull request merged**